src/librustdoc/html/render/write_shared.rs

```diff
@@ -408,9 +408,8 @@ pub(super) fn write_shared(
     // with rustdoc running in parallel.
     all_indexes.sort();
     write_crate("search-index.js", &|| {
-        let mut v = String::from("var searchIndex = JSON.parse('{\\\n");
-        v.push_str(&all_indexes.join(",\\\n"));
-        v.push_str("\\\n}');\ninitSearch(searchIndex);");
+        let v = static_files::SEARCH_JS
```

Comment:
Please don't make this change! `search-index.js` is only supposed to contain the search "content", nothing else. Create a new JS file or insert the content into `main.js` but don't change how we handle `search-index.js`.

Reply:
Ok! Happy to split this out into `search.js`. Why is `search-index.js` supposed to only contain the search content and nothing else?

Reply:
Multiple reason: this is a file different for every crate, meaning that docs.rs cannot store it once and for all. Also it's faster to parse a smaller file (even if here, the part you add might not be that big compared to the search index itself).

So here, it's mostly for docs.rs reason.

Reply:
And for docs.rs it would be nice to convert it to `.json` at some point :grin:

Reply:
Not possible as is unfortunately, you can load json files for local docs. :3